test/powershell/Language/Scripting/Array.Tests.ps1

```diff
@@ -0,0 +1,54 @@
+Describe "ArrayExpression Tests" -Tags "CI" {
+    It "@([object[]](1,2,3)) should return a 3-element array of object[]" {
+        $result = @([object[]](1,2,3))
+        $result.GetType().FullName | Should Be "System.Object[]"
+        $result.Length | Should Be 3
+    }
+
+    It "@([int[]](1,2,3)) should return a 3-element array of object[]" {
+        $result = @([int[]](1,2,3))
```

Comment:
For any array type other than `[object[]]`, it will fall in the code path of [`PSToObjectArrayBinder`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/parser/Compiler.cs#L5524). I just use `[int[]]` as a simple validation.